### PR TITLE
refactor: Deprecate the ability to configure aaa config when collecting show-techs

### LINKED ISSUE #894: Deprecate AAA configuration show collect show-tech

### PR DESCRIPTION
**Pull request by gmuloc**
# Description

ANTA should not configure the devices - only outputing a nice error message when something is missing

Fixes #894 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have run pre-commit for code linting and typing (`pre-commit run`)
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: anta/cli/exec/utils.py [hunk 1]
```diff
@@ -128,6 +128,10 @@ async def collect(device: AntaDevice) -> None:
                     logger.error("Unable to collect tech-support on %s: configuration 'aaa authorization exec default local' is not present", device.name)
                     return
 
+                # TODO: ANTA 2.0.0
+                msg = "[DEPRECATED] Using '--configure' for collecting show-techs is deprecated and will be removed in ANTA 2.0.0"
```
Comment (carl-baillargeon) [inline anta/cli/exec/utils.py:132]: Should we propose the alternative here (configure it yourself)?

File: anta/cli/exec/utils.py [hunk 2]
```diff
@@ -128,6 +128,10 @@ async def collect(device: AntaDevice) -> None:
...
+                logger.warning(msg)
```
Comment (carl-baillargeon) [inline anta/cli/exec/utils.py:133]: I think we should use `warnings.warn()` here so that users can control these warnings with `warnings.filterwarnings()`

Comment (gmuloc) [reply anta/cli/exec/utils.py:133]: no because by default warnings.warn deprecation warnings are disabled

Comment (gmuloc) [reply anta/cli/exec/utils.py:133]: https://docs.python.org/3/library/warnings.html#default-warning-filter

File: anta/cli/exec/utils.py (hunk 1 above)
Comment (gmuloc) [inline anta/cli/exec/utils.py:132]:
```suggestion
                msg = "[DEPRECATED] Using '--configure' for collecting show-techs is deprecated and will be removed in ANTA 2.0.0. Please add the required configuration on your devices before running this command. from ANTA."
```

**Review by carl-baillargeon — APPROVED** [review]
LGTM.

**Pull request merged**